src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -537,6 +549,66 @@ AppendShardIdToConstraintName(AlterTableCmd *command, uint64 shardId)
 
 
 /*
+ * UpdateWholeRowColumnReferencesWalker extends ColumnRef nodes that end with A_Star
+ * with the shardId contained in the given state.
+ *
+ * ColumnRefs that don't reference A_Star are not extended as catalog access isn't
+ * allowed here and we don't otherwise have enough context to disambiguate a
+ * field name that is identical to the table name.
+ */
+static bool
+UpdateWholeRowColumnReferencesWalker(Node *node, ColumnRefWalkerState *state)
```

Comment:
At this point, I feel this could just accept a `uint64 *` rather than declaring a struct with a single field.